### DISCUSSION TITLE
求大佬给个思路 这种连线如何做

### DISCUSSION THREAD

**Discussion by PZ-18664918826** [antvis/X6#2398] [Q&A] [answered]
鼠标移入节点显示连接桩，移出节点隐藏连击桩 求思路

**Comment by zmcode** [comment, marked as answer]
`function showPorts(ports, show) {
    for (let i = 0, len = ports.length; i < len; i = i + 1) {
        ports[i].style.visibility = show ? "visible" : "hidden";
    }
}

function  registerNodeShowPorts(graphInstance, id) {
    graphInstance.on("node:mouseenter", () => {
        const container = document.getElementById(id);
        const ports = container.querySelectorAll(".x6-port-body");
        showPorts(ports, true);
    });

    graphInstance.on("node:mouseleave", () => {
        const container = document.getElementById(id);
        const ports = container.querySelectorAll(".x6-port-body");
        showPorts(ports, false);
    });
}`

**Reply by PZ-18664918826** [reply]
感谢

**Comment by wb-wenbei** [comment]
参考下这个示例
https://x6.antv.vision/zh/examples/showcase/practices#flowchart

**Reply by PZ-18664918826** [reply]
感谢
